src/sdk/tracing/index.ts

```diff
@@ -253,8 +302,39 @@ export const startTracing = (options: InitializeOptions) => {
         }
       }
     }
+    // OVERRIDE_PARENT_SPAN makes the current span the root span in the trace.
+    // The backend looks for this attribute and deletes the parentSpanId if the
+    // attribute is present. We do that to make the topmost non-Next.js span
+    // the root span in the trace.
+    if (span.parentSpanId 
+        && nextSpanIds.has(span.parentSpanId) 
+        && !options.preserveNextJsSpans
+    ) {
+      span.setAttribute(OVERRIDE_PARENT_SPAN, true);
+    }
+    originalOnStart(span, parentContext);
+    // If we know by this time that this span is created by Next.js,
+    // we add it to the set of nextSpanIds.
+    if (isNextSpan(span) && !options.preserveNextJsSpans) {
+      addNextSpanId(span.spanContext().spanId);
+    }
+  };
+
+  _spanProcessor.onEnd = (span: ReadableSpan) => {
+    // Sometimes we only have know that this span is a Next.js only at the end.
+    // We add it to the set of nextSpanIds in that case. Also, we do not call
+    // the original onEnd, so that we don't send it to the backend.
+    if ((isNextSpan(span) || nextSpanIds.has(span.spanContext().spanId)) && !options.preserveNextJsSpans) {
+      addNextSpanId(span.spanContext().spanId);
+    } else {
+      // By default, we call the original onEnd.
+      originalOnEnd(span);
```

Comment:
why is it in the else block? Shouldn't we always call it, similar to how you do on line 315?

Reply:
Nope, because this is exactly what controls whether we send the span at all. See the comment in 325-326